src/server/controllers/model.controller.ts

```diff
@@ -1325,6 +1328,57 @@ export async function getModelTemplateFieldsHandler({
   }
 }
 
+export async function getModelTemplateFromBountyHandler({
+  input,
+  ctx,
+}: {
+  input: GetByIdInput;
+  ctx: DeepNonNullable<Context>;
+}) {
+  try {
+    const { id: userId } = ctx.user;
+    const awardedEntry = await dbRead.bountyEntry.findFirst({
+      where: { bountyId: input.id, benefactors: { some: { userId } } },
+      select: { id: true, bounty: true },
+    });
+
+    if (!awardedEntry) {
+      throw throwNotFoundError(`You have no awarded entries on the bounty with id ${input.id}`);
+    }
+
+    const { bounty } = awardedEntry;
+
+    if (!constants.bounties.supportedBountyToModels.some((t) => t === bounty.type)) {
+      throw throwBadRequestError('This bounty type is not supported for model creation');
+    }
+
+    const meta = bounty.details as BountyDetailsSchema;
+    const files = await getFilesByEntity({ id: awardedEntry.id, type: 'BountyEntry' });
+
+    const bountyTypeModelTypeMap: Record<string, ModelType> = {
+      [BountyType.ModelCreation]: ModelType.Checkpoint,
+      [BountyType.LoraCreation]: ModelType.LORA,
+    };
```

Comment:
Let's declare this outside of this method to avoid instantiating it every time this handler gets called